Doc/library/pickle.rst

```diff
@@ -30,9 +30,19 @@ avoid confusion, the terms used here are "pickling" and "unpickling".
 
 .. warning::
 
-   The :mod:`pickle` module is not secure against erroneous or maliciously
-   constructed data.  Never unpickle data received from an untrusted or
-   unauthenticated source.
+   The ``pickle`` module **is not secure**. Only unpickle data you trust.
+
+   It is possible to construct malicious pickle data which will **execute
+   arbitrary code during unpickling**. Never unpickle data that could have come
+   from an untrusted source, or that could have been tampered with.
+
+   Consider signing data with :mod:`hmac` if you need to ensure that it has not
+   been tampered with.
+
+   The JSON data interchange format provided by the :mod:`json` module is more
```

Comment:
I recommend leaving this out.  It goes well past giving an informative warning and moves into editorializing.  Also, for many uses of pickle, JSON is no substitute.

Reply:
I have changed this sentence to be less discursive and simply point to the `json` module and the "Comparison with `json`" section lower down.